lib/eventboss/unit_of_work.rb

```diff
@@ -12,12 +12,27 @@ def initialize(client, queue, listener, message)
       @listener = listener
       @message = message
       @logger = logger
+
+      listener_instance = @listener.new
```

Comment:
I'd rather see this part in some other place. You create a new instance of listener here, and then again in the `#run` - why not do it outside of this class and pass the ready Listener to UnitOfWork?